### ISSUE TITLE
A general question regarding dev and eval results

### ISSUE THREAD

**Issue by RoyJames** [kaldi-asr/kaldi#4717]
Hi,

I've used Kaldi for several benchmarks but still don't fully understand the role of dev and test sets here. In other deep learning domains, normally the train set is fine-tuned on the dev set, and the best performing model on the dev set is tested on the eval set. This makes the model slightly "overfitted" to the dev set, and may have a slight performance decrease when run on the test set. 

However, when I read Kaldi's code (I'm reading the `ami` recipe), the TDNN model is trained for a fixed number of epochs on the `train` split, and I don't see separate treatment for `dev` and `eval` sets during decoding. Then I'd expect these two results to be approximately the same since both of them are NOT used during training. But what I observe is `dev` WER are always better than `eval`. I'm confused why this is the case here. Am I misunderstanding the code or the data?

Thanks,
Zhenyu

**Comment by danpovey** [comment]
Whether dev or eval is better is likely due to how the dev and eval sets were selected.
The fact that we select the best LM-scale on each test set individually, in the scoring script, is definitely not "right" from a formal point of view, but it's likely not the biggest source of experimental error and typically it will affect all results in the same way so as long as you're comparing like with like it's not really a big deal.
I have generally not been very careful about separating the roles of dev and eval.  In many speech datasets there is so little test data anyway that it can make more sense to just always compute both.  And some datasets only have one or the other.
